### ISSUE TITLE
Running puppet cause a 'new' instance of es to start every time

### ISSUE THREAD

**Issue by jecnua** [elastic/puppet-elasticsearch#534]
After upgrading to 0.10.1 from 0.9.9 of this module I noticed that every time I run puppet, a new instance of es is started, LEAVING the old one running.

Ubuntu 14.04
This module version 0.10.1
ElasticSearch version to manage 2.1.1

This is what happens if I run ti twice:

```
$ ps aux | grep elastic
elastic+  8486  1.7 42.3 14514548 6964492 ?    Sl   Dec21  18:21 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
elastic+  8546  1.4 42.3 14512632 6956960 ?    Sl   Dec21  15:49 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   22555  0.0  0.0  10464   936 pts/0    R+   10:35   0:00 grep --color=auto elastic
```

As you can notice there is the same reference to a pid file:

```
$ cat /var/run/elasticsearch-my-domain-name.pid
8546
```
That obviously refer to only one of the two.
I tried to rollback to 0.9.9 but I had to give up becayse the plugin installation broke (since we know it changed).

I can't unfortunately paste the debug puppet run since there is too much sensible data in there, but please if there is any way to help you guys pinpoint this problem please let me know.

Regards,
F.

**Comment by electrical** [comment]
Hi,

That's very interesting. I run all kinds of acceptance tests on different distro's, including Ubuntu 14.04, and haven't seen this issue.
Can you reproduce the issue by using the init script it self?
The puppet module depends on the init script to send back the status and stuff.

**Comment by jecnua** [comment]
Hi,

Yes I tried manually all permutation of restart, stop, status I could think of and it looks pretty... erratic.

This is the case of me killing the processes and try to start one manually:

```
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
ubuntu   23325  0.0  0.0  10460   936 pts/0    S+   11:07   0:00 grep --color=auto elastic
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is not running
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name start
 * Starting Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                          [ OK ]
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is not running
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is not running
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
elastic+ 23377  222  5.3 14423756 885496 ?     Sl   11:07   0:11 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -javaagent:/opt/appdynamics/javaagent/javaagent.jar -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   23439  0.0  0.0  10460   936 pts/0    S+   11:07   0:00 grep --color=auto elastic
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name start
 * Starting Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                                  * Already running.                                                                                                                                                                                                                                                                                   [ OK ]
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is running
```

**Comment by electrical** [comment]
Hmm, interesting.. Could you kill all instances, remove the pid file(s) and try again?
Not sure if the init script gets confused about something.. I must admit i haven't seen this issue happen on any of my tests.

**Comment by jecnua** [comment]
I delete the pid file and the wrapper was looking pretty stable...

```
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
elastic+ 23377 10.6  7.2 14515596 1187844 ?    Sl   11:07   0:45 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   23559  0.0  0.0  10460   936 pts/0    S+   11:14   0:00 grep --color=auto elastic
ubuntu@elasticsearch-my-domain-name:~$ sudo kill 23377
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
ubuntu   23569  0.0  0.0   8728   392 pts/0    R+   11:14   0:00 grep --color=auto elastic
ubuntu@elasticsearch-my-domain-name:~$ ls -la /var/run/elasticsearch-elasticsearch-my-domain-name.pid
-rw-r--r-- 1 elasticsearch elasticsearch 5 Dec 22 11:07 /var/run/elasticsearch-elasticsearch-my-domain-name.pid
ubuntu@elasticsearch-my-domain-name:~$ cat /var/run/elasticsearch-elasticsearch-my-domain-name.pid
23377ubuntu@elasticsearch-my-domain-name:~$ sudo rm /var/run/elasticsearch-elasticsearch-my-domain-name.pid
ubuntu@elasticsearch-my-domain-name:~$ ls -la /var/run/elasticsearch-elasticsearch-my-domain-name.pid
ls: cannot access /var/run/elasticsearch-elasticsearch-my-domain-name.pid: No such file or directory

ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name start
 * Starting Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                          [ OK ]
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is not running
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
elastic+ 23632  197  6.4 14487588 1064800 ?    Sl   11:17   0:21 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   23728  0.0  0.0  10460   932 pts/0    S+   11:17   0:00 grep --color=auto elastic
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is running
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is running
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name restart
 * Stopping Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                          [ OK ]
 * Starting Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                          [ OK ]
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
elastic+ 23822  135  2.5 13932376 416648 ?     Sl   11:18   0:02 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   23849  0.0  0.0  10460   932 pts/0    S+   11:18   0:00 grep --color=auto elastic
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is running
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name start
 * Starting Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                                  * Already running.                                                                                                                                                                                                                                                                                   [ OK ]
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
elastic+ 23822  125  7.1 14517360 1168952 ?    Sl   11:18   0:23 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   23965  0.0  0.0  10460   936 pts/0    S+   11:18   0:00 grep --color=auto elastic
```

But...

```
ubuntu@elasticsearch-my-domain-name:~$ sudo puppet agent -t
[...]
ubuntu@elasticsearch-my-domain-name:~$ ps aux | grep elastic
elastic+ 25082  136  6.4 14489644 1058760 ?    Sl   11:19   0:24 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
elastic+ 25141  136  6.4 14487556 1052632 ?    Sl   11:19   0:23 /usr/bin/java -Xms10g -Xmx10g -Djava.awt.headless=true -XX:+UseG1GC -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.security.manager.enabled=false -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elasticsearch-2.1.1.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-elasticsearch-my-domain-name.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs=/var/log/elasticsearch/elasticsearch-my-domain-name -Des.default.path.data=/var/lib/elasticsearch-elasticsearch-my-domain-name -Des.default.path.work=/tmp/elasticsearch-elasticsearch-my-domain-name -Des.default.path.conf=/etc/elasticsearch/elasticsearch-my-domain-name
ubuntu   25374  0.0  0.0  10464   936 pts/0    S+   11:19   0:00 grep --color=auto elastic
```

**Comment by electrical** [comment]
That is very confusing :-( I'll see what i can reproduce with my integration tests against 2.1.1 and let you know.

**Comment by jecnua** [comment]
Any help is appreciated. I will try to dig into it again. If you cannot reproduce it feel free to close this ticket, maybe it's somehow connected to my specific environment (it happens to all the nodes in the cluster I tried it however).

Regards,
F.

**Comment by electrical** [comment]
Tried to reproduce it with my integration tests which do a few restarts but no luck so far. Very confusing.

**Comment by jecnua** [comment]
Thank you for trying. I will try to dig into the code if this module myself, maybe I can find it where this happens. Unfortunately the debug run doesn't help.

**Comment by jecnua** [comment]
Found it! Is all about time!
If I start elasticsearch in one shell and run a restart in another soon after, I have two instances.
I found it by noticing that in the wrapper if you call:
> xxx start
and soon after
> xxx status
will say 'not-running'.

I you can check it in the code I pasted in this thread yesterday:

```
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name start
 * Starting Elasticsearch Server elasticsearch-my-domain-name                                                                                                                                                                                                                          [ OK ]
ubuntu@elasticsearch-my-domain-name:~$ sudo service elasticsearch-elasticsearch-my-domain-name status
 * elasticsearch is not running
```

So if you have two calls close together that try a restart (example a restart and a puppet refresh) it will create two istances.
Before you ask why I do that, it's because the plugin installation is not really working in this module. There is no 'update' concept and it broke the license during the upgrade. So I delete the plugins, reinstall it and force a restart (these are all exec). if this happens during a 'puppet refresh', we get into my situation.

**Comment by electrical** [comment]
Ahhh okay.
With ES 2.x the plugin versions are directly coupled to the ES version.
I've built in the automatic upgrade but has only been implemented since version 0.10.0 of the puppet module.
So perhaps you could get rid of the custom thing?

Example of the auto upgrade: https://github.com/elastic/puppet-elasticsearch/blob/master/spec/acceptance/021_es2x_spec.rb#L165-L212

**Comment by jecnua** [comment]
I was using 0.10.1 when I implemented the 'custom thing' (let's call it for what it is... hack).
But I always received this:

```
ERROR: Plugin [marvel-agent] is incompatible with Elasticsearch
[2.1.1]. Was designed for version [2.0.0]
```
I can try again to see if it does update but I am not so optimist. 

PS: I install es via deb option, not repo.

**Comment by electrical** [comment]
How did you do the plugin puppet parts? I'm guessing there is an error there.

**Comment by jecnua** [comment]
```
  elasticsearch::plugin{'cloud-aws':
    instances  => $es_node_name,
    module_dir => 'cloud-aws' #NOTE: This is needed here
  }
  elasticsearch::plugin{'license':
    instances  => $es_node_name
  }->
  elasticsearch::plugin{'marvel-agent':
    instances  => $es_node_name
  }
  elasticsearch::plugin{'lmenezes/elasticsearch-kopf/2.0':
    instances  => $es_node_name
  }
```

**Comment by electrical** [comment]
Hmm, interesting.. the tests i did with ```elasticsearch::plugin{'cloud-aws': instances => 'es-01' }```
worked fine so far in my tests.
What it does on the puppet run is find out what the elasticsearch version is and if its not correct, uninstall the plugin.
Would you be able to do a --debug run to see what the plugin parts give back? it should say what ES version it found and stuff.

**Comment by jecnua** [comment]
Sorry for the delay I needed to fix the last node before getting a new one to try the debug: :)
Here:

```
Debug: Executing '/usr/share/elasticsearch/bin/elasticsearch -v'
Debug: Executing '/usr/share/elasticsearch/bin/elasticsearch --version'
Debug: Elasticsearch_plugin[cloud-aws](provider=plugin): Found ES version 2.1.1
Debug: Elasticsearch_plugin[cloud-aws](provider=plugin): Got cloud-aws Expected cloud-aws/2.1.1. Removing for reinstall
Debug: Executing '/usr/share/elasticsearch/bin/plugin remove cloud-aws'
Debug: Elasticsearch_plugin[cloud-aws](provider=plugin): Commands: ["install", ["cloud-aws"]]
Debug: Executing '/usr/share/elasticsearch/bin/plugin install cloud-aws'
Error: Execution of '/usr/share/elasticsearch/bin/plugin install cloud-aws' returned 64: -> Installing cloud-aws...
Trying https://download.elastic.co/elasticsearch/release/org/elasticsearch/plugin/cloud-aws/2.1.1/cloud-aws-2.1.1.zip ...
Downloading ............................................................................................................................................................................................................................................................................................................................................DONE
Verifying https://download.elastic.co/elasticsearch/release/org/elasticsearch/plugin/cloud-aws/2.1.1/cloud-aws-2.1.1.zip checksums if available ...
Downloading .DONE
ERROR: Plugin [license] is incompatible with Elasticsearch [2.1.1]. Was designed for version [2.0.0]
Error: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[cloud-aws]/Elasticsearch_plugin[cloud-aws]/ensure: change from absent to present failed: Execution of '/usr/share/elasticsearch/bin/plugin install cloud-aws' returned 64: -> Installing cloud-aws...
Trying https://download.elastic.co/elasticsearch/release/org/elasticsearch/plugin/cloud-aws/2.1.1/cloud-aws-2.1.1.zip ...
Downloading ............................................................................................................................................................................................................................................................................................................................................DONE
Verifying https://download.elastic.co/elasticsearch/release/org/elasticsearch/plugin/cloud-aws/2.1.1/cloud-aws-2.1.1.zip checksums if available ...
Downloading .DONE
ERROR: Plugin [license] is incompatible with Elasticsearch [2.1.1]. Was designed for version [2.0.0]
Notice: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[license]/Elasticsearch_plugin[license]: Dependency Elasticsearch_plugin[cloud-aws] has failures: true
Warning: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[license]/Elasticsearch_plugin[license]: Skipping because of failed dependencies
Notice: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[marvel-agent]/Elasticsearch_plugin[marvel-agent]: Dependency Elasticsearch_plugin[cloud-aws] has failures: true
Warning: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[marvel-agent]/Elasticsearch_plugin[marvel-agent]: Skipping because of failed dependencies
Notice: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[lmenezes/elasticsearch-kopf/2.0]/Elasticsearch_plugin[lmenezes/elasticsearch-kopf/2.0]: Dependency Elasticsearch_plugin[cloud-aws] has failures: true
Warning: /Stage[main]/Wdelasticsearch::Install/Elasticsearch::Plugin[lmenezes/elasticsearch-kopf/2.0]/Elasticsearch_plugin[lmenezes/elasticsearch-kopf/2.0]: Skipping because of failed dependencies
```

**Comment by electrical** [comment]
Ahh. interesting.. I've only done tests with a single plugin so far, not multiple plugins when doing the upgrade.
Let me do that as well and see if i can reproduce.

**Comment by electrical** [comment]
I managed to reproduce it reliably with 2 plugins. I've only done tests so far with a single plugin.
I also found out why its behaving this way and now trying to find a decent solution.
Will keep you updated around this.

**Comment by jecnua** [comment]
Thank you :+1:

**Comment by jecnua** [comment]
Hi Richard, any update on this? Not to hurry you, just to know if for now I need to hack my way around it since I am getting closer to a deadline. :)

**Comment by electrical** [comment]
@jecnua no solid solution yet.
So the problem is that the plugin manager code of ES doesn't only looks at the plugin you are trying to manage ( install, remove, etc ) but also to all other installed plugins.
So if you upgrade ES from 2.0.0 to 2.0.1 it requires a removal of all the plugins first before installing any new ones.
This is because if any plugin is still there for 2.0.0 the plugin manager will complain about it.

The only solution is to do a removal of all the plugins prior to installing any plugin, which is a very ugly solution and hard to implement.
I did raise this issue internally and we are looking on isolating the action of the plugin manager to the plugin you try to manage.

**Comment by jecnua** [comment]
I understand. I will try to do that then, trying to avoid the double instance problem (it still happens when you restart es in a rapid succession).
I will look forward to remove my hack as soon as this is solved ;)

**Comment by electrical** [comment]
Hi,

I've implemented a solution at #563 and merged.
This should solve the issue until its decently fixed in ES it self.
I'm planning to do a release next week Monday.

**Comment by jecnua** [comment]
Thank you very much :)
It will allow me to remove an horrible exec from my module until we have a better solution.

F.

**Comment by electrical** [comment]
Hi,

This fix has been released today in version 0.10.3 Can you try it out and see if it solves your problem?
Thanks!

**Comment by jecnua** [comment]
I just went through a 2.2 update, so at the moment I have nothing to update to test it. Feel free to close this ticket, in case the problem is still there or present in another form I can open new one. Thank you for going all the way to solve this problem. :)

F.
